components/Navbar/LangSwitch.tsx

```diff
@@ -0,0 +1,77 @@
+'use client'
+
+import LangBtn from 'public/assets/icons/language.svg'
+import { i18n } from 'i18n/config'
+import { useState, useEffect } from 'react'
+import { useRouter, usePathname } from 'next/navigation'
+import { getCurrentLocale, generateNewUrl } from 'hooks'
+import clsx from 'clsx'
+
+export default function LangSwitch() {
+  const router = useRouter()
+  const pathname = usePathname()
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+  const [currentLanguage, setCurrentLanguage] = useState<any>(1)
+
+  useEffect(() => {
+    setCurrentLanguage(getCurrentLocale(pathname))
+  }, [])
+
+  const handleLanguageClick = (language) => {
+    setCurrentLanguage(language)
+    setIsOpen(false)
+    const newLocale = i18n.locales[language]
+    router.replace(generateNewUrl(pathname, newLocale))
+  }
+
+  return (
+    <div className="relative inline-block text-left">
+      <div className="ml-4 h-10 w-10">
+        <button
+          type="button"
+          className="text-grey-300 h-10 cursor-pointer"
+          aria-label="Choose language"
+          aria-haspopup="true"
+          aria-expanded={isOpen}
+          onClick={() => setIsOpen(!isOpen)}
+        >
+          <LangBtn className="h-6 w-6" />
+        </button>
+      </div>
+
+      <div
+        className={clsx(
+          'absolute right-0 mt-2 w-[125px] origin-top-right  border border-white border-opacity-25 bg-back shadow-lg',
+          {
+            block: isOpen,
+            hidden: !isOpen,
+          }
+        )}
+        aria-orientation="vertical"
+        aria-labelledby="language-menu"
+      >
+        <div className="py-1">
+          {i18n.languages.map((language, index) => (
```

Comment:
> "Want to help translate saving satoshi to more languages?"

We might want to add this option to the end of our locale list

Reply:
Perhaps this tooltip from the conversation in #148 could be in a "Your Language" option.
I am going to look around on some other websites to see if there is a common theme people use

Reply:
I have not addressed this feedback as the design/ plan for adding a contribution path for language is not clear yet.